### PR TITLE
ci: Add test execution on Pull Request

### LINKED ISSUE #95: [FEAT] :sparkles: Setup GitHub actions to run tests on every PR
### Description

we need to run tests on backend on every PR

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by alberturria**
### Description

Added a new github action workflow, so we ensure that the tests are run on pull request.

**Dependency**:
I've seen that there are at least a couple of sensitive environment variables that are hardcoded in the repo. That's why I'm making usage of the [encripted secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets#using-encrypted-secrets-in-a-workflow) within the workflow.
⚠️ This means that these needs to be added as a secret to the github project ⚠️ 
### Checklist:

- [x] This pull request follow our contribution guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (postman and swagger)
- [x] I have written tests for the code
- [x] I have updated .env.example file for new .env variables

### What is current behavior?

N/A

Closes #95

### What is new behavior?

The tests are ran on PR.

### Breaking Changes?

N/A

### Additional information

N/A

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: Tests
+on: [pull_request]
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    env:
+      GOOGLE_CLIENT_ID: ${{ secrets.GOOGLE_CLIENT_ID }}
+      GOOGLE_CLIENT_SECRET: ${{ secrets.GOOGLE_CLIENT_SECRET }}
```
Comment (aayushchugh) [inline .github/workflows/tests.yml:8]: Please include another env variables too

They are mentioned in `.env.example`

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    env:
+      GOOGLE_CLIENT_ID: ${{ secrets.GOOGLE_CLIENT_ID }}
+      GOOGLE_CLIENT_SECRET: ${{ secrets.GOOGLE_CLIENT_SECRET }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "16"
+      - run: npm i
+      - run: npm t
```
Comment (aayushchugh) [inline .github/workflows/tests.yml]: Use `yarn test` command to run tests

**Review by aayushchugh — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (alberturria) [reply .github/workflows/tests.yml:8]: Sure! ✅

File: .github/workflows/tests.yml (hunk 2 above)
Comment (alberturria) [reply .github/workflows/tests.yml]: Changed! 👍

**Review by aayushchugh — APPROVED** [review]

**Comment by aayushchugh** [comment]
@alberturria I will merge this once you mark this PR as `Ready for Review`

**Comment by alberturria** [comment]
> @alberturria I will merge this once you mark this PR as `Ready for Review`

Great! it should be no longer marked as draft 🙂

**Comment by aayushchugh** [comment]
@alberturria will merge this PR tomorrow morning

**Comment by aayushchugh** [comment]
@all-contributors please add @alberturria for code

**Pull request merged**